### PR TITLE
Add CPUSoftmax for model

### PR DESCRIPTION
**Pull request by KSTconsult**

### CODE REVIEW

**Comment by KSTconsult** [comment]
@l-bat @RodionovF @TolyaTalamanov  @apertovs

File: src/runtime/cpu/layers/cpusoftmax.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#include "runtime/cpu/layers/cpusoftmax.hpp"
+
+deepworks::cpu::CPUSoftmax::CPUSoftmax(deepworks::LayerInfo info)
+        : deepworks::cpu::ICPULayer(info) {
```
Comment (TolyaTalamanov) [inline src/runtime/cpu/layers/cpusoftmax.cpp:4]: `std::move`

File: src/runtime/cpu/layers/cpusoftmax.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                                       std::vector<deepworks::Tensor>& outputs) {
+    DeepWorks_Assert(inputs.size()  == 1u && "Softmax takes only one input");
+    DeepWorks_Assert(outputs.size() == 1u && "Softmax produce only one output");
+
+    DeepWorks_Assert(inputs.front().shape().size()  == 2u && "Softmax input must be 2D");
+    DeepWorks_Assert(outputs.front().shape().size() == 2u && "Softmax output must be 2D");
+}
+
+void deepworks::cpu::CPUSoftmax::forward(const std::vector<deepworks::Tensor>& inputs,
+                                      std::vector<deepworks::Tensor>& outputs) {
```
Comment (TolyaTalamanov) [inline src/runtime/cpu/layers/cpusoftmax.cpp:17]: alignment

File: src/runtime/cpu/layers/cpusoftmax.cpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    validate(inputs, outputs);
+
+    const auto& input  = inputs.front();
+    auto& output = outputs.front();
+
+    const auto& in_shape  = input.shape();
+    const auto& out_shape = output.shape();
+    deepworks::CPUSoftmaxForward({input.data() , in_shape[0] , in_shape[1]},
+                              {output.data(), out_shape[0], out_shape[1]});
```
Comment (TolyaTalamanov) [inline src/runtime/cpu/layers/cpusoftmax.cpp:26]: and here

File: tests/unit/test_cpusoftmax.cpp [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#include <deepworks/deepworks.hpp>
+
+#include "kernels_reference.hpp"
+#include "test_utils.hpp"
+
+namespace dw = deepworks;
+
+TEST(LayerTests, CPUSoftmax) {
+int n_features = 100;
```
Comment (TolyaTalamanov) [inline tests/unit/test_cpusoftmax.cpp:9]: alignment ?

**Review by TolyaTalamanov — APPROVED** [review]
LGTM :+
Great, thanks!

**Review by l-bat — APPROVED** [review]

File: src/runtime/cpu/layers/cpusoftmax.cpp [hunk 5]
```diff
@@ -0,0 +1,27 @@
+#include "runtime/cpu/layers/cpusoftmax.hpp"
+
+deepworks::cpu::CPUSoftmax::CPUSoftmax(deepworks::LayerInfo info)
```
Comment (l-bat) [inline src/runtime/cpu/layers/cpusoftmax.cpp:3]:
```suggestion
deepworks::cpu::CPUSoftmax::CPUSoftmax(deepworks::LayerInfo&& info)
```

**Pull request merged**